src/components/EditTodoTitle/EditTodoTitle.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  handleChangeTodo: <T>(
+    callback: (id: T) => Promise<void[]>,
+    valueToCall: T,
+  ) => void;
+  todo: Todo;
+  setIsEditing: (state: boolean) => void;
+  onUpdateTodo: (todo: Todo[]) => Promise<void[]>;
+  onDeleteTodo: (id: number[]) => Promise<void[]>;
+}
+
+export const EditTodoTitle: React.FC<Props> = ({
+  onDeleteTodo,
+  handleChangeTodo,
+  todo,
+  setIsEditing,
+  onUpdateTodo,
+}) => {
+  const [titleName, setTitleName] = useState(todo.title);
+
+  const handleChangeTitle = (newTitle: string) => {
+    setTitleName(newTitle);
+  };
+
+  const inputField = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    inputField.current?.focus();
+  }, [todo]);
```

Comment:
Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.